core/server/src/main/java/tachyon/master/file/FileSystemMasterRestServiceHandler.java

```diff
@@ -0,0 +1,398 @@
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    } catch (IOException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * @param path the file path
+   * @return the response object
+   */
+  @POST
+  @Path(NEW_BLOCK_ID_FOR_FILE)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response getNewBlockIdForFile(@QueryParam("path") String path) {
+    try {
+      return Response.ok(mFileSystemMaster.getNewBlockIdForFile(new TachyonURI(path))).build();
+    } catch (TachyonException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * @param path the path to remove
+   * @param recursive whether to remove paths recursively
+   * @return the response object
+   */
+  @POST
+  @Path(REMOVE)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response remove(@QueryParam("path") String path,
+                         @QueryParam("recursive") boolean recursive) {
+    try {
+      mFileSystemMaster.deleteFile(new TachyonURI(path), recursive);
+      return Response.ok().build();
+    } catch (TachyonException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    } catch (IOException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * @param srcPath the source path
+   * @param dstPath the destination path
+   * @return the response object
+   */
+  @POST
+  @Path(RENAME)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response remove(@QueryParam("srcPath") String srcPath,
+                         @QueryParam("dstPath") String dstPath) {
+    try {
+      mFileSystemMaster.rename(new TachyonURI(srcPath), new TachyonURI(dstPath));
+      return Response.ok().build();
+    } catch (TachyonException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    } catch (IOException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * @param path the file path
+   * @return the response object
+   */
+  @GET
+  @Path(SCHEDULE_ASYNC_PERSIST)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response scheduleAsyncPersist(@QueryParam("path") String path) {
+    try {
+      return Response.ok(mFileSystemMaster.scheduleAsyncPersistence(new TachyonURI(path))).build();
+    } catch (TachyonException e) {
+      return Response.serverError().status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
+
+  /**
+   * @param path the file path
+   * @param owner the file owner user name
+   * @param group the file owner group name
+   * @param permission the file permission bits
+   * @param recursive whether to set permissions recursively
+   * @return the response object
+   */
+  @PUT
+  @Path(SET_ACL)
+  @Produces(MediaType.APPLICATION_JSON)
```

Comment:
If all the methods produce JSON, then you can put this annotation at class level.